src/compute/src/server.rs

```diff
@@ -204,6 +206,10 @@ impl mz_cluster::types::AsRunnableWorker<ComputeCommand, ComputeResponse> for Co
         persist_clients: Arc<PersistClientCache>,
         tracing_handle: Arc<TracingHandle>,
     ) {
+        if config.context.worker_core_affinity {
+            set_core_affinity(timely_worker.index());
```

Comment:
`set_core_affinity` handles the case when we run in multi-process replicas. However, it might be nicer if we explicitly compute the process-local worker ID here instead of relying on `set_core_affinity` to do the right thing?

Reply:
The nice thing about using the module approach instead of trying to be more explicit using knowledge about the number of workers per process is that it also works in dev/CI environments where all replica processes share the same set of CPUs.

Say you are running a '2-2' replica on your laptop, core pinning works just fine if you have more than four cores. If we used process-local worker IDs instead, cores 0 and 1 would each be double-assigned.